src/System.Private.ServiceModel/tools/test/SelfHostWcfService/CertificateResources/UserCertificateResource.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using WcfTestBridgeCommon;
+
+namespace WcfService.CertificateResources
+{
+    internal class UserCertificateResource : EndCertificateResource
+    {
+        public UserCertificateResource() : base() { }
+
+        // Requests a certificate to be generated by the Bridge based on a user name and not machine name
+        public override ResourceResponse Put(ResourceRequestContext context)
+        {
+            X509Certificate2 certificate;
+
+            string subject; 
+            if (!context.Properties.TryGetValue(subjectKeyName, out subject) || string.IsNullOrWhiteSpace(subject))
+            {
+                throw new ArgumentException("When PUTting to this resource, specify an non-empty 'subject'", "context.Properties");
+            }
+
+            // There can be multiple subjects, separated by ,
+            string[] subjects = subject.Split(',');
+
```

Comment:
Hardened this a little bit - comment still stands that we could do a lot more, but not in scope for this PR.